### PR TITLE
Fix readme 4 r10 k

### PR DESCRIPTION
**Pull request by gapotts**
This a pull request for issue 254 https://github.com/theforeman/puppet-puppet/issues/254.

Added verbiage around supporting R10K with Directory environment.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,11 @@ Puppet 3.6+ and config environments on older versions.  These are set up under
 /etc/puppet/environments/ - change `server_environments` to define the list to
 create, or use `puppet::server::env` for more control.
 
+When using directory environments with R10K you need to set the 
+`server_environments` parameter to an empty array ie. \[] to prevent
```
Comment (mmoll) [inline README.md]: `[]` instead of \[] is maybe better.

File: README.md [hunk 2]
```diff
@@ -23,6 +23,11 @@ Puppet 3.6+ and config environments on older versions.  These are set up under
 /etc/puppet/environments/ - change `server_environments` to define the list to
 create, or use `puppet::server::env` for more control.
 
+When using directory environments with R10K you need to set the 
+`server_environments` parameter to an empty array ie. \[] to prevent 
+`r10k deploy environments` from reporting an error cause by the creation of top
```
Comment (mmoll) [inline README.md]: "caus_ed_"?

File: README.md [hunk 3]
```diff
@@ -96,6 +101,15 @@ wrapper classes or even your ENC (if it supports param classes). For example:
       server_reports        => 'store',
       server_external_nodes => '',
     }
+    
+    # The same example as above but overiding `server_environment` for R10K
```
Comment (mmoll) [inline README.md]: server_environment_s_

File: README.md [hunk 4]
```diff
@@ -96,6 +101,15 @@ wrapper classes or even your ENC (if it supports param classes). For example:
...
+    class { '::puppet':
+      server                => true,
+      server_foreman        => false,
+      server_reports        => 'store',
+      server_external_nodes => '',
+      server_environment    => [],
```
Comment (mmoll) [inline README.md]: also with s here

**Comment by mmoll** [comment]
The wording could be a bit more non-r10k general, but I'm also fine with it as-is, please address the line comments and squash all the commits into one at the end. Thanks :)

**Comment by gapotts** [comment]
By squashed I'm assume you are referring to the line break between the two comment under the same heading, correct?

**Comment by mmoll** [comment]
http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit - At the end only one single commit should be in this pull request, containing all your changes.

File: README.md [hunk 5]
```diff
@@ -96,6 +99,15 @@ wrapper classes or even your ENC (if it supports param classes). For example:
       server_reports        => 'store',
       server_external_nodes => '',
     }
+    
+    # The same example as above but overiding `server_environments` for R10K
```
Comment (mmoll) [inline README.md]: over_r_iding

**Comment by gapotts** [comment]
Ok thanks for the URL. I'll squash them into one commit, after I fix the the override.

**Comment by gapotts** [comment]
Ok, Made the changes and squashed the commits. Let me know if there is any thing else you need me to do.

Regard- gapotts

File: README.md [hunk 6]
```diff
@@ -96,6 +99,15 @@ wrapper classes or even your ENC (if it supports param classes). For example:
...
       server_external_nodes => '',
     }
+    
+    # The same example as above but overriding `server_environments` for R10K
+    class { '::puppet':
+      server                => true,
+      server_foreman        => false,
+      server_reports        => 'store',
+      server_external_nodes => '',
+      server_environments    => [],
```
Comment (mmoll) [inline README.md]: one space too much :space_invader:

**Comment by gapotts** [comment]
Duh! I should of caught that.

**Comment by mmoll** [comment]
:+1:

**Comment by ekohl** [comment]
:+1:

**Comment by domcleal** [comment]
Thanks @gapotts!

**Comment by gapotts** [comment]
No problem.

**Pull request merged**